spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/AbstractSpannerIntegrationTest.java

```diff
@@ -0,0 +1,203 @@
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.DatabaseId;
+import com.google.cloud.spanner.Spanner;
+import org.junit.After;
+import org.junit.Before;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.gcp.autoconfigure.spanner.GcpSpannerAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.spanner.GcpSpannerProperties;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntity;
+import org.springframework.cloud.gcp.data.spanner.core.mapping.SpannerPersistentEntityImpl;
+import org.springframework.cloud.gcp.data.spanner.repository.config.EnableSpannerRepositories;
+import org.springframework.cloud.gcp.data.spanner.test.AbstractSpannerIntegrationTest.TestConfiguration;
+import org.springframework.cloud.gcp.data.spanner.test.domain.Trade;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.data.util.TypeInformation;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * <p>
+ * For Spanner integration tests, you will need to have an instance predefined, everything
+ * else is generated. The instance by default is "integration-instance", which can be
+ * overriden in <code>src/test/resources/application-test.properties</code>, the property:
+ * <code>test.integration.spanner.instance</code>
+ * </p>
+ * <p>
+ * Within the <code>integration-instance</code> instance, there is a single database for
+ * tests, <code>integration-db</code>. This is automatically created, if doesn't exist.
+ * The tables are generated to have a unique postfix, which is updated on the entity
+ * annotations as well dynamically to avoid collisions of multiple parallel tests running
+ * against the same instance.
+ * </p>
+ *
+ * @author Balint Pato
+ */
+@SpringBootTest(classes = { TestConfiguration.class, GcpSpannerAutoConfiguration.class })
+public abstract class AbstractSpannerIntegrationTest {
```

Comment:
After reading the Javadoc and class name, I'm still not quite too sure what this class does or why we need it.

I suspect it just initializes the Spanner environment that the specific tests can then use. In that case, we could be a little more to the point on the Javadoc, but I won't make a fuss about that.

Reply:
Sounds good, I'll add more explanation.